app_build_suite/build_steps/helm.py

```diff
@@ -730,6 +734,114 @@ def cleanup(
+    """
+
+    @property
+    def steps_provided(self) -> Set[StepType]:
+        return {STEP_VALIDATE}
+
+    def initialize_config(self, config_parser: configargparse.ArgParser) -> None:
+        config_parser.add_argument(
+            "-g",
+            "--enable-giantswarm-helm-validator",
+            required=False,
+            default=True,
+            action="store_true",
+            help="Should Giant Swarm specific validation be enabled",
+        )
+        config_parser.add_argument(
+            "-s",
+            "--enable-strict-giantswarm-validator",
+            required=False,
+            default=True,
+            action="store_true",
+            help="If strict mode is enabled, the build fails when any validation rule fails; otherwise, a WARN is "
+            "given",
+        )
+        config_parser.add_argument(
+            "--giantswarm-validator-ignored-checks",
+            required=False,
+            default="",
+            help="Comma-separated list of Giant Swarm validation checks to ignore even if they fail",
+        )
+
+    def pre_run(self, config: argparse.Namespace) -> None:
+        """Runs a set of Giant Swarm specific validations."""
+        if not config.enable_giantswarm_helm_validator:
+            logger.debug("Not running Giant Swarm specific chart validation.")
+            return
+
+        gs_validators = self._load_giant_swarm_validators()
+
+        ignore_list: List[str] = []
+        ignore_str_list: List[str] = config.giantswarm_validator_ignored_checks.split(",")
+        for name in ignore_str_list:
+            n = name.strip()
+            if n:
+                ignore_list.append(n)
+
+        for validator in gs_validators:
+            validator_name = type(validator).__name__
+            logger.info(f"Running Giant Swarm validator '{validator.get_check_code()}: {validator_name}'.")
+            if validator.validate(config):
+                logger.debug(f"Giant Swarm validator '{validator.get_check_code()}: {validator_name}' is OK.")
+            else:
+                msg = f"Giant Swarm validator '{validator.get_check_code()}: {validator_name}' failed its checks."
+                if config.enable_strict_giantswarm_validator and validator.get_check_code() not in ignore_list:
+                    raise ValidationError(self.name, msg)
+                else:
+                    logger.warning(msg)
+
+    def _load_giant_swarm_validators(self) -> List[GiantSwarmValidator]:
+        gs_validators: List[GiantSwarmValidator] = []
+        current_frame = inspect.currentframe()
+        if current_frame is None:
+            raise ValidationError(self.name, "Can't check current frame and detect current module's path.")
+
+        cur_mod_path = os.path.dirname(os.path.abspath(inspect.getfile(current_frame)))
+        validators_mod_path = os.path.join(cur_mod_path, "giant_swarm_validators")
+        for modname in listdir(validators_mod_path):
```

Comment:
Why do we dynamically load the validators here? I feel like having just imports or something simpler would be sufficient. Basically I don't understand the usecase for loading them dynamically like this.